tests/test_other.py

```diff
@@ -1289,18 +1290,14 @@ def tmp_source(name, code):
     # of 'libA.a' after 'libB.a' adds undefined symbol "x", so a.c.o will now be
     # included (and the link will succeed).
     libs = ['-Wl,--start-group'] + libs_list + ['-Wl,--end-group']
-    output = run_process([PYTHON, EMCC] + args + libs, stdout=PIPE, stderr=PIPE)
-    out_js = 'a.out.js'
-    assert os.path.exists(out_js), output.stdout + '\n' + output.stderr
```

Comment:
Don't need/want to assert these exist?
Is the output stdout/stderr printed if the emcc command fails?

Reply:
run_process doesn't redirect stdout stderr and in general we don't want to when calling emcc because it should mostly be silent.

For tests that go "compiler + run" I think we can assume that if compile succeeds the output file will be created and just try to run it.. the run step will still fail if the file is not there, but since it extremely unlikely that emcc will succeed but *not* write the output file I don't think it worth littering all our run_process calls to EMCC with extra assertions.  Better to let the process return code signal success. 

If you don't agree we should create a wrapper "run_emcc" to and use ot everywhere that will both check the return code and check for the existence of the file.

Reply:
I agree with that, I'm just not sure how I feel about not having output here in the event that things go wrong. Still it's probably reasonable enough to debug that I think this is fine in the tradeoff-space.

Reply:
In almost all the tests we have when emcc fails you get a subprocess failed to return 0 exception, and you need to look at the stderr to see why.  I don't see why this test should be different.